src/api/scenarioData/scenarioData.controller.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  Controller,
+  Get,
+  Param,
+  Post,
+  Body,
+  Put,
+  Delete,
+  HttpException,
+  HttpStatus,
+  Res,
+  UploadedFile,
+  UseInterceptors,
+} from '@nestjs/common';
+import { FileInterceptor } from '@nestjs/platform-express';
+import { Response } from 'express';
+import { ScenarioDataService } from './scenarioData.service';
+import { ScenarioData } from './scenarioData.schema';
+import { CreateScenarioDto } from './dto/create-scenario.dto';
+
+@Controller('api/scenario')
+export class ScenarioDataController {
+  constructor(private readonly scenarioDataService: ScenarioDataService) {}
+
+  @Get('/list')
+  async getAllScenarioData(): Promise<ScenarioData[]> {
+    try {
+      return await this.scenarioDataService.getAllScenarioData();
+    } catch (error) {
+      throw new HttpException('Error getting all scenario data', HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  @Get('/by-rule/:ruleId')
+  async getScenariosByRuleId(@Param('ruleId') ruleId: string): Promise<ScenarioData[]> {
+    try {
+      return await this.scenarioDataService.getScenariosByRuleId(ruleId);
+    } catch (error) {
+      throw new HttpException('Error getting scenarios by rule ID', HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
You might want to update these HttpExceptions so that they don't only ever return INTERNAL_SERVER_ERROR. Because sometimes its not an internal error, but the rule just doesn't exist. See "runDecisionByFile" in "decisions.service" for an example.